pkg/api/write.go

```diff
@@ -260,6 +261,9 @@ func ingest(
 			ingestor.FinishWriteRequest(req)
 			return false
 		}
+		// set number of received samples and metadata in case below Ingest fails
+		numSamples = uint64(getTotalSamples(req))
+		numMetadata = uint64(len(req.Metadata))
```

Comment:
Won't this be always be overwritten by the Ingest below? Am I missing something?

Reply:
Good catch!

Reply:
Funny thing is that I added a comment stating that we prevent in case of `Ingest` failing but accidentally put it before the error is returned :) Note to myself: don't work on PRs when it's a public holiday :)